src/components/Search/SearchComponent.tsx

```diff
@@ -0,0 +1,134 @@
+import { forwardRef, useCallback, useEffect, useImperativeHandle, useRef, useState } from "react";
+import { motion } from "framer-motion";
+import { MagnifyingGlass, X } from "@phosphor-icons/react";
+import { Link, useNavigate } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+
+export type SearchComponentHandler = {
+    toggle: () => void;
+}
+
+export const SearchComponent = forwardRef<SearchComponentHandler>((_, ref) => {
+    const { t } = useTranslation();
+  
+  const [isOpen, setIsOpen] = useState(false);
+  const [historic, setHistoric] = useState<Array<string>>([]);
```

Comment:
pls also change other types to [] and historic to history
```suggestion
  const [history, setHistory] = useState<string[]>([]);
```